### PR TITLE
Fixed Bug For DocumentComment in v2_3

### PR DESCRIPTION
**Pull request by neilnaveen**
Concerning issue #186 

- added DocumentComment switch case for tag in parse_creation_info.go
- added test for DocumentComment switch case
- added file parse_document.go for the switch case above
- added test for parse_document.go corrosponding func in v2_3
- fixed typo in v2_2's DocumentComment test func's name made in PR #185 

Signed-off-by: Neil Naveen <42328488+neilnaveen@users.noreply.github.com>

### CODE REVIEW

File: spdx/v2/v2_3/tagvalue/reader/parse_document.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+func (parser *tvParser) parsePairForDocument(tag string, value string) error {
+	if parser.doc == nil {
+		return fmt.Errorf("no document struct created in parser ann pointer")
+	}
+
+	switch tag {
+	case "DocumentComment":
+		parser.doc = &v2_3.Document{}
```
Comment (kzantow) [inline spdx/v2/v2_3/tagvalue/reader/parse_document.go:15]: I'm not sure this is the right thing to do here -- if `parser.doc` is already set, all values will be overwritten with defaults

File: spdx/v2/v2_3/tagvalue/reader/parse_creation_info_test.go [hunk 2]
```diff
@@ -425,3 +425,14 @@ func TestFailsExternalDocumentReferenceWithInvalidFormats(t *testing.T) {
 		}
 	}
 }
+func TestParserCIMovesToOtherLicenseAfterParsingDocument(t *testing.T) {
```
Comment (kzantow) [inline spdx/v2/v2_3/tagvalue/reader/parse_creation_info_test.go:428]: This appears to have the same naming issue as the other PR, and I don't see a change updating the other one

**Review by kzantow — CHANGES_REQUESTED** [review]

File: spdx/v2/v2_3/tagvalue/reader/parse_creation_info.go [hunk 3]
```diff
@@ -94,6 +94,8 @@ func (parser *tvParser) parsePairFromCreationInfo(tag string, value string) erro
 		return parser.parsePairForAnnotation(tag, value)
 	case "AnnotationComment":
 		return parser.parsePairForAnnotation(tag, value)
+	case "DocumentComment":
+		return parser.parsePairForDocument(tag, value)
```
Comment (kzantow) [inline spdx/v2/v2_3/tagvalue/reader/parse_creation_info.go:98]:
```suggestion
		return parser.parsePairFromStart(tag, value)
```

And get rid of the `parsePairForDocument` function.

**Comment by kzantow** [comment]
@neilnaveen do you think you will be able to get to the aforementioned changes in the next few days? We would like to get this done fairly soon due to the `main` branch currently having the bug mentioned here

**Comment by neilnaveen** [comment]
> @neilnaveen do you think you will be able to get to the aforementioned changes in the next few days? We would like to get this done fairly soon due to the `main` branch currently having the bug mentioned here. If you don't have time for it, I would be happy to help -- just let me know!

I will try to work on it this week.

File: spdx/v2/v2_3/tagvalue/reader/parse_creation_info.go (hunk 3 above)
Comment (neilnaveen) [reply spdx/v2/v2_3/tagvalue/reader/parse_creation_info.go]: Fixed, thanks!

File: spdx/v2/v2_3/tagvalue/reader/parse_creation_info_test.go (hunk 2 above)
Comment (neilnaveen) [reply spdx/v2/v2_3/tagvalue/reader/parse_creation_info_test.go:428]: Fixed, thanks!

File: spdx/v2/v2_3/tagvalue/reader/parse_document.go (hunk 1 above)
Comment (neilnaveen) [reply spdx/v2/v2_3/tagvalue/reader/parse_document.go]: Fixed, thanks!

**Review by kzantow — APPROVED** [review]
Great, thanks @neilnaveen !

**Pull request merged**